libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2877,7 +2876,16 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 					if (from.location() == DataLocation::Memory)
 						body = "converted := value";
 					else if (from.location() == DataLocation::CallData)
-						solUnimplemented("Conversion of calldata types not yet implemented.");
+						body = Whiskers(R"(
+							converted := <allocateMemoryArray>(<length>)
+							<copyToMemory>(value, converted, mul(<length>, <sizeInBytes>))
+						)")
+						("functionName", functionName)
+						("allocateMemoryArray", allocateMemoryArrayFunction(to))
+						("copyToMemory", copyToMemoryFunction(from.location() == DataLocation::CallData))
```

Comment:
Can you really use a raw copy in all situations?

Reply:
The more complicated situations probably need a loop and another call to the conversion function.

Reply:
This works only for byte arrays and arrays of value type (both dynamic and static)

Other situations I plan to do as a follow up. Also, other situations are not implemented in old codegen as well.